### PR TITLE
Move no-old-flexbox to table formatter

### PR DESCRIPTION
**Pull request by ebidel**
R: @patrickhulce all

This PR:
- add `<pre>` support to the markdown renderer
- fixes a bug in the `no-old-flexbox` audit where it was not finding all old flexbox uses. The audit was incorrectly testing vendor prefixed css properties and matching them with prefixed values. They need to be run separately.
- adds some smokehouse checks to cache all the pokemon
- moves the audit to use the table formatter.
- attributes inline styles to "inline". Issue in #1500.

screen shot 2017-01-26 at 11 31 11 am

### CODE REVIEW

**Comment by brendankenny** [comment]
> to cache all the pokemon

so close :P

File: lighthouse-core/audits/dobetterweb/no-old-flexbox.js [hunk 1]
```diff
@@ -53,28 +53,42 @@ class NoOldFlexboxAudit extends Audit {
...
+    const sheetsUsingOldFlexbox = displayPropResults.concat(otherPropResults);
+
     const urlList = [];
-    sheetsUsingDisplayBox.forEach(sheet => {
+    sheetsUsingOldFlexbox.forEach(sheet => {
       sheet.parsedContent.forEach(props => {
         const formattedStyleRule = StyleHelpers.getFormattedStyleRule(sheet.content, props);
+
+        let url = sheet.header.sourceURL;
+        if (!url) {
```
Comment (patrickhulce) [inline lighthouse-core/audits/dobetterweb/no-old-flexbox.js]: I also check if it matches the pageUrl if we want to be consistent. Also thoughts on using getDisplayName here?

**Comment by patrickhulce** [comment]
I have a fix in #1536 for the wonky indentation of the I can split out too btw

**Comment by ebidel** [comment]
caching pokemon is important for great performance

File: lighthouse-core/lib/styles-helpers.js [hunk 2]
```diff
@@ -96,14 +96,14 @@ function getFormattedStyleRule(content, parsedContent) {
     }, []).join('\n');
   }
 
-  const block = `
-${parsedContent.selector} {
+  const block = `\`\`\`${parsedContent.selector} {
```
Comment (brendankenny) [inline lighthouse-core/lib/styles-helpers.js]: doing this in the `extendedInfo` value seems a bit ugly. What about a different keyword for the table formatter that annotates code blocks with the ```s?

File: lighthouse-core/lib/styles-helpers.js [hunk 3]
```diff
@@ -96,14 +96,14 @@ function getFormattedStyleRule(content, parsedContent) {
...
   ${rule}
-}`;
+}\`\`\``;
 
   return {
     styleRule: block.trim(),
-    location: `line: ${startLine}, row: ${start}, col: ${end}`
+    startLine,
+    location: `${start}:${end}`
```
Comment (brendankenny) [inline lighthouse-core/lib/styles-helpers.js]: we should think about settling on a format for parsed source blocks at some point. ESTree has a semi-standard format (albeit for JS): https://github.com/estree/estree/blob/master/es5.md#node-objects

File: lighthouse-core/audits/dobetterweb/no-old-flexbox.js (hunk 1 above)
Comment (ebidel) [inline lighthouse-core/audits/dobetterweb/no-old-flexbox.js]: done

File: lighthouse-cli/test/fixtures/dobetterweb/dbw_tester.css [hunk 4]
```diff
@@ -18,3 +18,6 @@
 body {
   background-color: #eee;
 }
+.doesnotapply {
+  display: flexbox; /* FAIL */
+}
```
Comment (ebidel) [inline lighthouse-cli/test/fixtures/dobetterweb/dbw_tester.css]: done

File: lighthouse-core/audits/dobetterweb/no-old-flexbox.js [hunk 5]
```diff
@@ -53,28 +53,42 @@ class NoOldFlexboxAudit extends Audit {
...
+          url = 'inline';
+        }
+
         urlList.push({
-          url: sheet.header.sourceURL,
-          label: formattedStyleRule.location,
+          url,
```
Comment (ebidel) [inline lighthouse-core/audits/dobetterweb/no-old-flexbox.js]: added in the `url === pageUrl` check above

File: lighthouse-core/lib/styles-helpers.js (hunk 2 above)
Comment (ebidel) [inline lighthouse-core/lib/styles-helpers.js]: done

File: lighthouse-core/lib/styles-helpers.js (hunk 3 above)
Comment (ebidel) [inline lighthouse-core/lib/styles-helpers.js]: ack

**Comment by ebidel** [comment]
PTAL

**Comment by brendankenny** [comment]
needs a rebase due to table churn

**Comment by ebidel** [comment]
rebased!

File: lighthouse-core/formatters/table.js [hunk 6]
```diff
@@ -86,6 +86,8 @@ class Table extends Formatter {
         switch (key) {
           case 'code':
             return '`' + value.trim() + '`';
+          case 'pre':
+            return '\`\`\`\n' + result[key].trim() + '\`\`\`';
```
Comment (brendankenny) [inline lighthouse-core/formatters/table.js]: add this to `table-formatter-test.js`?

Comment (brendankenny) [inline lighthouse-core/formatters/table.js]: oh, and added to the jsdoc above :)

Comment (ebidel) [inline lighthouse-core/formatters/table.js]: done

**Pull request merged**